mem3_helper.py

```diff
@@ -72,12 +73,36 @@ def connect_the_dots(names):
             time.sleep(5)
             resp = requests.put(uri, data=json.dumps(doc))
         print('Adding cluster member', name, resp.status_code)
+    print('Cluster membership populated!')
 
-def sleep_forever():
+def check_membership_forever():
+    creds = (os.getenv("COUCHDB_USER"), os.getenv("COUCHDB_PASSWORD"))
+    expected_peers_count = os.getenv("COUCHDB_CLUSTER_SIZE")
+    uri = "http://127.0.0.1:5984/_membership"
+    print('Checking _membership data...')
     while True:
-        time.sleep(5)
+        if creds[0] and creds[1]:
+            resp = requests.get(uri, auth=creds)
+        else:
+            resp = requests.get(uri)
```

Comment:
I think I'd extract this block and the identical block in `connect_the_dots()` to its own function accepting `uri` as a parameter and returning the response object.

Reply:
They are not identical. Block in `connect_the_dots()` sends data using put method, and this block sends no data and just uses get method. I tried to move it into separate function and ended up with even more code, so I would prefer to keep this as it is.